### PR TITLE
[build.sh] Automate creating GitHub releases

### PR DESCRIPTION
**Pull request by segiddins**
Closes most of https://github.com/realm/realm-cocoa-private/issues/65

\c @jpsim

### CODE REVIEW

File: scripts/github_release.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+BUILD_SH = Pathname(__FILE__).+('../../build.sh').expand_path
+VERSION = `sh '#{BUILD_SH}' get-version`.strip
+RELEASE = "v#{VERSION}"
+
+BUILD = ARGV[0] ? Pathname(ARGV[0]).expand_path : BUILD_SH.parent + 'build'
+OBJC_ZIP = BUILD + "realm-objc-#{VERSION}.zip"
+SWIFT_ZIP = BUILD + "realm-swift-#{VERSION}.zip"
+CARTHAGE_ZIP = BUILD + 'Carthage.framework.zip'
+
+FileUtils.cp SWIFT_ZIP, CARTHAGE_ZIP
```
Comment (jpsim) [inline scripts/github_release.rb]: The Carthage zip isn't just a copy of the Swift zip. Here's how I make the Carthage zip with bash:

```
mkdir -p Carthage/Build/Mac Carthage/Build/iOS
mv realm-swift-0.93.1/ios/Realm.framework realm-swift-0.93.1/ios/RealmSwift.framework Carthage/Build/iOS
mv realm-swift-0.93.1/osx/Realm.framework realm-swift-0.93.1/osx/RealmSwift.framework Carthage/Build/Mac
zip --symlinks -r Carthage.framework.zip Carthage
```

File: scripts/github_release.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+github = Octokit::Client.new
+github.access_token = ENV['GITHUB_ACCESS_TOKEN']
+
+puts 'Creating GitHub release'
+response = github.create_release(REPOSITORY, RELEASE, name: RELEASE, body: RELEASE_NOTES)
+release_url = response[:url]
+
+uploads = [OBJC_ZIP, SWIFT_ZIP, CARTHAGE_ZIP]
+uploads.each do |upload|
+  puts "Uploading #{upload.basename} to Github"
```
Comment (jpsim) [inline scripts/github_release.rb]: GitHub

**Comment by jpsim** [comment]
:+1: looks like it does *all* of https://github.com/realm/realm-cocoa-private/issues/65 to me.

**Comment by segiddins** [comment]
Well, jenkins needs to be setup to actually call `sh build.sh github-release`

**Comment by jpsim** [comment]
Jenkins should now run the [`create_github_release_cocoa`](https://ci.realm.io/view/Release%20Cocoa/job/create_github_release_cocoa/) as part of the release_cocoa job now.

**Pull request merged**